### PR TITLE
Add bundlesize package and tweak travis config

### PR DESCRIPTION
**Pull request by nerrad**
This pull:

- adds the [bundlesize package](https://github.com/siddharthkp/bundlesize) and implements npm scripts for it.
- adds a new `build:ci` script for running build/linting/bundlesize in a ci environment.
- tweaks travis config so that it has named builds and splits up things a bit more.

### Bundlesize

An increasing conversation we're going to be having is the size of our built bundles in the context of client frontend loading.  I think it would be good for us to start monitoring the sizes of our builds both to ensure we fall within a max size we set as a target and to aid with immediately getting feedback if sizes significantly increase or decrease in pulls.  I was motivated in part to do this pull because I want to experiment with various options for avoiding duplicating component logic between frontend and backend so this will help (in part) to monitor changes in bundle sizes while experimenting (Note: I'm aware of the caveat that this does _not_ capture any external bundles sizes (wp.components etc)... but my experiments don't involve them 😉 )

The bundlesize library is fairly established, is one of the libraries mentioned in p7H4VZ-2hW-p2 , and we can look at connecting it to repo so that we can get updates about bundlesize posted in pull checks:

screenshot

### Travis config

Right now the travis config installs a full WordPress environment on every build when it's only really needed for phpunit tests.  Travis-ci has a job stages and named builds feature that we can utilize to split things up a bit so that:

- we only do full `composer install` and `WordPress install` on builds that need it.
- split php tests/linting from javascript tests/linting for clearer cause of build failures (which aids with quicker inspection and resolution).
- Assuming we get permissions, bundlesize also runs as a part of the javascript lint build and will report to the related pull check (as illustrated above).

**Update:** As noted in the thread of this pull, doing this configuration change exposed that phpunit tests were actually failing.  Initially the reveal was that the tests weren't even building in the travis environment.  I got that fixed but now we're seeing failing tests (which I'm able to reproduce locally as well).

### CODE REVIEW

**Comment by nerrad** [comment]
@mikejolley and @Aljullu in this pull I split out phpunit tests so they are separated from js tests and in the process exposed that the current phpunit run on travis fails.  At first I thought this may be something introduced in this pull but then I noticed that the previous configuration ran both phpunit and js tests in the same build and it looks like phpunit tests have been failing in other builds too (example here: https://travis-ci.com/woocommerce/woocommerce-gutenberg-products-block/jobs/223619204).  So I think I can take care of fixing phpunit tests running in travis-ci in this pull.

Another thing @mikejolley, I'd like to auth https://github.com/siddharthkp/bundlesize so we can trigger bundlesize checks in our pulls.  I see you're an owner on the @woocommerce org so is that something we can do?  We can sync up next time we're both online and get that done (or discuss why not if that's not an option)?

**Comment by nerrad** [comment]
Ok I fixed the fatal in the phpunit tests (see 441eb00) but now there are failing phpunit tests :).  So that's the next task here.  Unclear yet whether the failing tests are due to changed expectations somewhere along the way or actual fails (I suspect the former).

**Comment by nerrad** [comment]
Some more findings:

- If I run phpunit in the blocks folder locally with a _built_ version of woocommerce (i.e. installed from the latest release zip), tests pass.
- If I run phpunit in the blocks folder locally with a development version of woocommerce (setup the same as how it's setup via travis: running `npm install` and `composer install`), then I get the same fails as travis.  I don't know if this is an artifact of what's currently on master for woocommerce or not yet, but at least I can reproduce locally.

**Comment by nerrad** [comment]
All green, ready for review

File: tests/bootstrap.php [hunk 1]
```diff
@@ -19,65 +19,101 @@
...
 			exit( 1 );
 		}
+	}
+	return $dir;
+}
+
+/**
+ * Install WC Blocks
+ */
+function wc_blocks_install() {// Clean existing install first.
```
Comment (Aljullu) [inline tests/bootstrap.php]: Can you add a space here?
```suggestion
function wc_blocks_install() { // Clean existing install first.
```

File: tests/bootstrap.php [hunk 2]
```diff
@@ -19,65 +19,101 @@
...
+tests_add_filter(
+	'muplugins_loaded',
+	function() {
+		wc_load_core();
+		// install blocks plugin
+		wc_blocks_install();
 	}
 );
 
+tests_add_filter('setup_theme','wc_install_core');
```
Comment (Aljullu) [inline tests/bootstrap.php]: Should this be padded with spaces?
```suggestion
tests_add_filter( 'setup_theme', 'wc_install_core' );
```

File: tests/bootstrap.php (hunk 1 above)
Comment (nerrad) [reply tests/bootstrap.php]: doh, this is a comment that was supposed to be deleted instead (used to be other code here that was moved elsewhere).

Comment (nerrad) [reply tests/bootstrap.php]: fixed in f3049e0

File: tests/bootstrap.php (hunk 2 above)
Comment (nerrad) [reply tests/bootstrap.php]: fixed in f3049e0 and d2dee63

**Comment by nerrad** [comment]
Since:

- this is mostly introducing changes to testing tools.
- The two locations modified in our code (346f4b8 and 50db988) were done with @mikejolley's input.

I'm going to go ahead and merge this after tests pass.

**Pull request merged**